src/component/component.js

```diff
@@ -254,6 +254,30 @@ function normalizeOptions<P, X, C>(options : ComponentOptionsType<P, X, C>) : No
     };
 }
 
+/**
+ * Modify the decorate properties definition function to run once only.
+ * Currently we have a problem about overriding function reference if we called
+ * setProps twice, and now that is the case with the new container argument.
+ * 
+ * @param opts the defined options in the component creation step
+ * @return the same opts object with decorate function modified to run once if exists
+ */
+function decorateRunOnceModifier(opts : ComponentOptionsType<P, X, C>): ComponentOptionsType<P, X, C> {
```

Comment:
This approach seems reasonable to me! Two things come to mind:
1. Let's use `Object.keys()` instead of `Object.values()` and `Object.entries` since its safe to use with the current browser support requirements.
2. Let's flatten out this code to improve readability. Also, I'm wondering if we can call `decorateRunOnceModifier()` inside of `normalizeOptions` so it's all in one spot.

Example of flattening out the code:
```js
function decorateRunOnceModifier(opts : ComponentOptionsType<P, X, C>): ComponentOptionsType<P, X, C> {
    if (!isObject(opts) {
        return opts;
    }

    if (!isObject(opts.props)) {
        return opts;
    }

    Object.values(opts.props).forEach(parentValue => {
        if (!isObject(parentValue)) {
            return;
        }

        Object.entries(parentValue).forEach(([key, value]) => {
            if (key === "decorate") {
                parentValue[key] = once(value);
            }
        });
    });


    return opts;

}

```

Reply:
This solution won't apply